### PR TITLE
Add WindowsDiscoverConfig to find JDK

### LINKED ISSUE #25368: JDK is not discovered on Windows
When running SBT on Windows, it did not find the JDK:

```
[error] java.util.NoSuchElementException: key not found: 8
[error]         at scala.collection.immutable.Map$EmptyMap$.apply(Map.scala:98)
[error]         at scala.collection.immutable.Map$EmptyMap$.apply(Map.scala:96)
[error]         at akka.AkkaBuild$.$anonfun$defaultSettings$6(AkkaBuild.scala:104)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:39)
[error]         at sbt.std.Transform$$anon$4.work(System.scala:66)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:263)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[error]         at sbt.Execute.work(Execute.scala:272)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:263)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:174)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]         at java.lang.Thread.run(Thread.java:748)
```

The JDK was manually installed in a custom location. The `JAVA_HOME` environment variable correctly pointed to it, so perhaps we should take into account the `JAVA_HOME` or look at the `PATH` when discovering JDK's.

### PR DESCRIPTION
**Pull request by richardimaoka**
Fixes  #25368

I **think** the standard JDK installation location on Windows is `"C://Program Files/Java"`, as the default location chosen by Oracle JDK is this:

2018-07-27_20h44_51

2018-07-27_20h45_44

Previously, I had the same error as in the linked issue,

```
[error] java.util.NoSuchElementException: key not found: 8
[error]         at scala.collection.immutable.Map$EmptyMap$.apply(Map.scala:98)
[error]         at scala.collection.immutable.Map$EmptyMap$.apply(Map.scala:96)
[error]         at akka.AkkaBuild$.$anonfun$defaultSettings$6(AkkaBuild.scala:104)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:39)
[error]         at sbt.std.Transform$$anon$4.work(System.scala:66)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:263)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[error]         at sbt.Execute.work(Execute.scala:272)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:263)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:174)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]         at java.lang.Thread.run(Thread.java:748)
```

but after doing this change, I got akka compiled on Widows.

### CODE REVIEW

**Comment by richardimaoka** [comment]
If this is ok, should we also make a similar change in sbt?

Refs: https://github.com/akka/akka/pull/24711/files#r194035659

File: project/CrossJava.scala [hunk 1]
```diff
@@ -117,6 +117,17 @@ object CrossJava {
             }
     }
 
+    class WindowsDiscoverConfig extends JavaDiscoverConf {
+      val base: File = file("C://Program Files/Java")
```
Comment (johanandren) [inline project/CrossJava.scala]: Isn't there also a different app directory for 64-bit installations on windows?

Comment (richardimaoka) [reply project/CrossJava.scala]: I think `C://Program Files` is ok as follows. I couldn't find the official Microsoft resource so let me paste this random article, but as far as I checked this seems correct.

https://www.howtogeek.com/129178/why-does-64-bit-windows-need-a-separate-program-files-x86-folder/

> In summary, on a 32-bit version of Windows, you just have a “C:\Program Files” folder. This contains all your installed programs, all of which are 32-bit.

> On a 64-bit version of Windows, 64-bit programs are stored in the “C:\Program Files” folder and 32-bit programs are stored in the “C:\Program Files (x86)” folder.

Comment (johanandren) [reply project/CrossJava.scala]: Ah ok, that's how they do it, kinda makes sense (I have just seen it at some point on someone else machine, haven't run windows for ages). Thanks for verifying.

**Review by johanandren — APPROVED** [review]

File: project/CrossJava.scala [hunk 2]
```diff
@@ -117,6 +117,17 @@ object CrossJava {
...
+      val JavaHomeDir = """jdk-?(1\.)?([0-9]+).*""".r
+
+      def javaHomes: Vector[(String, File)] =
+        wrapNull(base.list())
+          .collect {
+            case dir@JavaHomeDir(m, n) => JavaVersion(nullBlank(m) + n).toString -> (base / dir)
```
Comment (ktoso) [inline project/CrossJava.scala]: usually space around ` @ `

**Review by ktoso — APPROVED** [review]
LGTM Thanks !

**Pull request merged**